atom/lib/import-js.js

```diff
@@ -166,100 +166,84 @@ function askForSelections(selectionsToAskFor, previousSelections) {
 }
 
 /**
- * @param {String} selections
+ * @param {Object} result
  */
-function importWord(selections) {
-  const args = [];
-
-  if (selections) {
-    // We have selections to import.
-    args.push('--selections', selections);
-  } else {
-    // There are no selections to import, so let's figure out what the current
-    // word is.
-    const word = getCurrentWord();
-    if (!word) {
-      return;
-    }
-    args.push('--word', word);
+function processResultMessages(result) {
+  if (result.messages) {
+    // TODO come up with a better solution for this potential wall of text that
+    // can happen when fixing a lot of imports.
+    atom.notifications.addSuccess(result.messages);
   }
+}
 
-  exec(args)
-    .then((result) => {
-      if (result.ask_for_selections) {
-        askForSelections(result.ask_for_selections, []);
-        return;
-      }
-
-      const editor = atom.workspace.getActiveTextEditor();
-      if (editor.getText() !== result.output) {
-        editor.buffer.setTextViaDiff(result.output);
-      }
-
-      if (result.messages) {
-        atom.notifications.addSuccess(result.messages);
-      }
-    })
-    .catch((error, result) => {
-      const message = result ? result.error.output : error.toString();
-      atom.notifications.addError(message);
-      throw error;
-    });
+/**
+ * @param {Object} result
+ */
+function handleExecResult(result) {
+  if (result.ask_for_selections) {
+    askForSelections(result.ask_for_selections, []);
+    return;
+  }
+
+  const editor = atom.workspace.getActiveTextEditor();
+  if (editor.getText() !== result.output) {
+    editor.buffer.setTextViaDiff(result.output);
+  }
+
+  processResultMessages(result);
 }
 
-function gotoWord() {
+/**
+ * @param {Error} error
+ * @param {Object} result
+ */
+function handleExecError(error, result) {
+  const message = result ? result.error.output : error.toString();
+  atom.notifications.addError(message);
+  throw error;
+}
+
+function importWord() {
   const word = getCurrentWord();
   if (!word) {
     return;
   }
 
-  exec(['--word', word, '--goto'])
-    .then((result) => {
-      const path = `${getCurrentWorkingDirectory()}/${result.output}`;
-      atom.open({ pathsToOpen: [path], newWindow: false });
+  exec(['--word', word])
+    .then(handleExecResult)
+    .catch(handleExecError);
+}
 
-      if (result.messages) {
-        atom.notifications.addSuccess(result.messages);
-      }
-    })
-    .catch((error, result) => {
-      const message = result ? result.error.output : error.toString();
-      atom.notifications.addError(message);
-      throw error;
-    });
+function fixImports() {
+  exec()
+    .then(handleExecResult)
+    .catch(handleExecError);
 }
 
 /**
  * @param {String} selections
  */
-function fixImports(selections) {
-  const args = [];
-  if (selections) {
-    args.push('--selections', selections);
+function importSelections(selections) {
+  exec(['--selections', selections])
```

Comment:
Are you sure? Here's [my response to your comment](https://github.com/trotzig/import-js/pull/231#discussion_r56357271).